Filter files by suffix

KochTobi opened the pull request:
Includes the filter option for both comands: `list` and `download`.

KochTobi committed:
Filter by suffix

Steffengreiner commented on src/main/java/life/qbic/model/download/QbicDataDownloader.java:
```diff
@@ -101,142 +99,75 @@ private static Path getTopDirectory(Path path) {
-            sampleCode, DEFAULT_DOWNLOAD_ATTEMPTS);
-        downloadFiles(downloadRequest);
+        List<DataSetFile> files = qbicDataFinder.getFiles(permID, fileFilter);
+        if (files.isEmpty()) {
+          LOG.info("Nothing to download for dataset " + sampleDataset.getPermId() + ".");
```
Nit: Is this relevant information concerning suffix filtering? 
Dependent on the suffix this can spam the log if a project has a multitude of datasets 

`15:07:02.631 [main] INFO  life.qbic.model.download.QbicDataDownloader - Nothing to download for dataset 20220923142207884-971611.`

Steffengreiner reviewed (dismissed):
Heya @KochTobi Nicely done, I tested all possible permutations concerning the suffixes and can confirm that it works as intended 👍

KochTobi committed:
Remove `Initializing download ...` message

Removed the message as it does not provide any information that is of use for the users. It leads to confusion in case no files matching the suffixes are found.
